Conceptpower+Spring/src/main/java/edu/asu/conceptpower/validation/UserValidator.java

```diff
@@ -0,0 +1,69 @@
+package edu.asu.conceptpower.validation;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.validation.Errors;
+import org.springframework.validation.Validator;
+
+import edu.asu.conceptpower.users.IUserManager;
+import edu.asu.conceptpower.users.User;
+
+@Component
+public class UserValidator implements Validator {
+    @Autowired
+    IUserManager usersManager;
+
+    @Override
+    public boolean supports(Class<?> arg0) {
+        return User.class.isAssignableFrom(arg0);
+    }
+
+    @Override
+    public void validate(Object arg0, Errors err) {
+
+        User user = (User) arg0;
+        String username = user.getUsername();
+        String fullName = user.getName();
+        String emailid = user.getEmail();
+        String pw = user.getPw();
+
+        // Validator for username - reject if empty or if contains special
+        // characters.
+        if (username == null || !username.matches("^[a-z0-9_-]{3,16}$")) {
+            
+            err.rejectValue("username", "required.username");
```

Comment:
You're validating two different things here:
1) if there is a username at all
2) does the username match requirments

There should be two different error messages for these scenarios, otherwise the user gets an error message saying "Username is required" although they provided it, which wouldn't help at all.

Also, you probably want to include A-Z as well in the regex, no reason to disallow uppercase letters, is there?